src/Fantomas.Tests/UtilsTests.fs

```diff
@@ -112,3 +113,65 @@ SetupTesting.generateSetupScript __SOURCE_DIRECTORY__
 #endif
 """
     |> mergeAndCompare a b
+
+[<Test>]
+let ``when input is empty`` () =
+    let property (p: bool) : bool =
+        let before, after = List.partitionWhile (fun _ _ -> p) []
+        before = [] && after = []
+
+    Check.QuickThrowOnFailure(property true)
+    Check.QuickThrowOnFailure(property false)
+
+[<Test>]
+let ``when predicate always returns false`` () =
+    let property (xs: int list) : bool =
+        let before, after =
+            List.partitionWhile (fun _ _ -> false) xs
+
+        before = [] && after = xs
+
+    Check.QuickThrowOnFailure property
+
+[<Test>]
+let ``when predicate always returns true`` () =
+    let property (xs: int list) : bool =
+        let before, after =
+            List.partitionWhile (fun _ _ -> true) xs
+
+        before = xs && after = []
+
+    Check.QuickThrowOnFailure property
+
+[<Test>]
+let ``when predicate returns true until certain index`` () =
+    let property (xs: int list, i: int) : bool =
+        let before, after =
+            List.partitionWhile (fun index _ -> i <> index) xs
+
+        let beforeLength = List.length before
+        let afterLength = List.length after
+
+        beforeLength = i
+        && afterLength = List.length xs - i
+        && before @ after = xs
+
+    let gen =
+        gen {
+            let! xs =
+                Arb.generate<int>
+                |> Gen.listOf
+                |> Gen.filter (fun l -> l.Length > 0)
+
+            let len = List.length xs
+
+            let! n =
+                Arb.generate<int>
+                |> Gen.filter (fun n -> n >= 0 && n < len)
```

Comment:
Similarly here, you can avoid a lot of filtering with `|> Gen.map abs` and getting rid of the `n >= 0` filter.

Reply:
In fact it turns out that `Gen.choice(0, len)` is a thing, so you can just directly construct what you need.

Reply:
Awesome. I need to spend some time reading the FsCheck docs now that I've got a taste for it.